api_v3/services/GroupUserService.php

```diff
@@ -431,21 +434,52 @@ protected function validateKuserkGroupCoExistence(kuser $kGroup, $kuserId)
 
 	protected function validateMaxGroupsPerUser($partnerId, $kuserId)
 	{
-		$criteria = new Criteria();
-		$criteria->add(KuserKgroupPeer::KUSER_ID, $kuserId);
-		$criteria->add(KuserKgroupPeer::STATUS, KuserKgroupStatus::ACTIVE);
+		$maxGroupsPerUser = $this->getPartnerMaximumGroupsPerUser($partnerId);
+
+		$kuserGroupList = $this->retrieveActiveGroupsForKuser($kuserId);
+		if (count($kuserGroupList) < $maxGroupsPerUser)
+		{
+			return;
+		}
+
+		$countLimitedGroups = $this->countLimitedGroups($kuserGroupList);
+		if ($countLimitedGroups >= $maxGroupsPerUser)
+		{
+			throw new KalturaAPIException (KalturaErrors::USER_EXCEEDED_MAX_GROUPS);
+		}
+	}
 
-		$maxGroupsPerUser = GroupUserService::DEFAULT_MAX_NUMBER_OF_GROUPS_PER_USER;
+	protected function getPartnerMaximumGroupsPerUser($partnerId)
+	{
 		$groupUserCountLimitMap = kConf::get('group_user_count_limit', 'local', array());
 		$partnersList = array_keys($groupUserCountLimitMap);
 		if (in_array($partnerId, $partnersList))
 		{
-			$maxGroupsPerUser = $groupUserCountLimitMap[$partnerId];
+			return $groupUserCountLimitMap[$partnerId];
 		}
 
-		if (KuserKgroupPeer::doCount($criteria) >= $maxGroupsPerUser)
+		return GroupUserService::DEFAULT_MAX_NUMBER_OF_GROUPS_PER_USER;
+	}
+
+	protected function retrieveActiveGroupsForKuser($kuserId)
+	{
+		$criteria = new Criteria();
+		$criteria->add(KuserKgroupPeer::KUSER_ID, $kuserId);
+		$criteria->add(KuserKgroupPeer::STATUS, KuserKgroupStatus::ACTIVE);
+		return KuserKgroupPeer::doSelect($criteria);
```

Comment:
Did you test a user with a lot of groups? 
As this now loads the entire object to run time, we need to make sure we won't get OOO

Reply:
how much would you say is a lot?